### PR TITLE
tools/lightning: batch doc updates for 2.1.6

### PR DESCRIPTION
**Pull request by dcalvin**
Batch doc updates for Lightning V2.1.6.
In alignment with PR: https://github.com/pingcap/docs/pull/945/

### CODE REVIEW

File: tools/lightning/csv.md [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+- 如果 `backslash-escape` 为 true，下列转义符会被识别并转换。 
+
+    | Sequence | Converted to             |
+    |----------|--------------------------|
+    | `\0`     | Null character (U+0000)  |
+    | `\b`     | Backspace (U+0008)       |
+    | `\n`     | Line feed (U+000A)       |
+    | `\r`     | Carriage return (U+000D) |
+    | `\t`     | Tab (U+0009)             |
+    | `\Z`     | Windows EOF (U+001A)     |
```
Comment (kennytm) [inline tools/lightning/csv.md]: Should this table be in Chinese?

File: tools/lightning/csv.md [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    ```
+
+- 当 `trim-last-separator = false`，该文件会被解析为包含 5 个字段的行 `('A', '', 'B', '', '')`。
+- 当 `trim-last-separator = true`，该文件会被解析为包含 3 个字段的行 `('A', '', 'B')`。
+
+### 不可配置项
+
+Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例如：
+
+* 行终止符只能是 CR（`\r`），LF（`\n`）或 CRLF （`\r\n`），也就是说，无法通过 `LINES TERMINATED BY` 自定义。
```
Comment (kennytm) [inline tools/lightning/csv.md]:
```suggestion
* 行终止符只能是 CR（`\r`），LF（`\n`）或 CRLF （`\r\n`），也就是说，无法自定义 `LINES TERMINATED BY`。
```

File: tools/lightning/deployment.md [hunk 3]
```diff
@@ -282,11 +288,24 @@ TiDB-Lightning 可随 TiDB 集群一起用 [Ansible 部署](../../op-guide/ansib
...
-    addr = "172.16.31.10:20170"
+    addr = "172.16.31.10:8287"
 
     [mydumper]
     # 文件读取区块大小。
     read-block-size = 65536 # 字节 (默认 = 64 KB)
+
+    #（源数据文件）单个导入区块大小的最小值。
+    # Lightning 根据该大小将一张大表分割为多个数据引擎文件。
+    batch-size = 107_374_182_400 # Byte (default = 100 GB)
```
Comment (kennytm) [inline tools/lightning/deployment.md]:
```suggestion
    batch-size = 107_374_182_400 # 字节 (默认 = 100 GB)
```

File: tools/lightning/deployment.md [hunk 4]
```diff
@@ -282,11 +288,24 @@ TiDB-Lightning 可随 TiDB 集群一起用 [Ansible 部署](../../op-guide/ansib
...
+    # 引擎文件要按序导入。因为是并行处理，多个数据引擎几乎同时被导入，
+    # 这样形成的处理队列会造成资源浪费。因此，Lightning 稍微增大了前几个
+    # 区块的大小，从而合理分配资源。该参数也决定了向上扩展（scale up）因
+    # 数，代表在完全并发下“导入”和“写入”过程的持续时间比。这个值也可以通过
+    # 计算 1 GB 大小单张表的（导入时长/写入时长）得到。精确的时间可以在日志
+    # 里看到。如果 “导入”更快，区块大小差异就会更小；比值为 0 则说明区块大小
+    # 是一致的。取汁范围是（0 <= batch-import-ratio < 1）。
```
Comment (kennytm) [inline tools/lightning/deployment.md]:
```suggestion
    # 是一致的。取值范围是（0 <= batch-import-ratio < 1）。
```

**Comment by dcalvin** [comment]
@kennytm  PTAL again. Thanks

File: tools/lightning/csv.md [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+separator = ','
+# 引用定界符，可以为 ASCII 字符或空字符。
+delimiter = '"'
+# CSV 文件是否包含表头。
+# 如果为 true，首行将会被跳过。
+header = true
+# CSV 是否包含 NULL。
+# 如果为 true，CSV 文件的任何列都不能解析为 NULL。
+not-null = false
+# 如果 `not-null` 为 false（CSV 可以包含 NULL），
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
# 如果 `not-null` 为 false（即 CSV 可以包含 NULL），
```

File: tools/lightning/csv.md [hunk 6]
```diff
@@ -0,0 +1,209 @@
...
+### `header`
+
+- 是否*所有* CSV 文件都包含表头行。
+- 如为 true，第一行会被用作*列名*。如为 false，第一行并无特殊性，按普通的数据行处理。
+
+### `not-null` 和 `null`
+
+- `not-null` 决定是否所有字段不能为空。
+- 如果 `not-null` 为 false，设定了 `null` 的字符串会被转换为 SQL NULL 而非具体数值。
+- 引用不会为 NULL 的字段。
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
- 引用不影响字段被解析为 NULL。
```

File: tools/lightning/csv.md [hunk 7]
```diff
@@ -0,0 +1,209 @@
...
+
+    例如有如下 CSV 文件：
+
+    ```csv
+    A,B,C
+    \N,"\N",
+    ```
+
+    在默认设置（`not-null = false; null = '\N'`）下，列 `A` and `B` 导入 TiDB 后都将会转换为 NULL。列 `C` 是空字符串，但并不解析为 NULL。
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
    在默认设置（`not-null = false; null = '\N'`）下，列 `A` and `B` 导入 TiDB 后都将会转换为 NULL。列 `C` 是空字符串 `''`，而不是 NULL。
```

File: tools/lightning/csv.md [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+    ```
+
+- 当 `trim-last-separator = false`，该文件会被解析为包含 5 个字段的行 `('A', '', 'B', '', '')`。
+- 当 `trim-last-separator = true`，该文件会被解析为包含 3 个字段的行 `('A', '', 'B')`。
+
+### 不可配置项
+
+Lightning 并不完全支持 `LOAD DATA` 语句中的所有配置项。例如：
+
+* 行终止符只能是 CR（`\r`），LF（`\n`）或 CRLF （`\r\n`），也就是说，无法自定义 `LINES TERMINATED BY`。
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
* 行终止符只能是 CR（`\r`），LF（`\n`）或 CRLF（`\r\n`），也就是说，无法自定义 `LINES TERMINATED BY`。
```

File: tools/lightning/csv.md [hunk 9]
```diff
@@ -0,0 +1,209 @@
...
+* 不可使用行前缀 （`LINES STARTING BY`）。
+* 不可跳过表头（`IGNORE n LINES`）。如有表头，必须是有效的列名。
+* 定界符和分隔符只能为单个 ASCII 字符。 
+
+## 通用配置
+
+### CSV
+
+默认设置已按 RFC 4180 调整。
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
默认设置已按照 RFC 4180 调整。
```

File: tools/lightning/deployment.md [hunk 10]
```diff
@@ -61,6 +62,8 @@ category: tools
 - `-F 256`：将每张表切分成多个文件，每个文件大小约为 256 MB。
 - `--skip-tz-utc`：添加这个参数则会忽略掉 TiDB 与导数据的机器之间时区设置不一致的情况，禁止自动转换。
 
+如果数据源是 CSV 文件，请参考 [CSV 支持](../../tools/lightning/csv.md] 获取配置信息。
```
Comment (yikeke) [inline tools/lightning/deployment.md]:
```suggestion
如果数据源是 CSV 文件，请参考 [CSV 支持](../../tools/lightning/csv.md]获取配置信息。
```

File: tools/lightning/deployment.md [hunk 11]
```diff
@@ -282,11 +288,24 @@ TiDB-Lightning 可随 TiDB 集群一起用 [Ansible 部署](../../op-guide/ansib
...
+
+    #（源数据文件）单个导入区块大小的最小值。
+    # Lightning 根据该大小将一张大表分割为多个数据引擎文件。
+    batch-size = 107_374_182_400 # 字节 (默认 100 GB)
+    # 引擎文件要按序导入。因为是并行处理，多个数据引擎几乎同时被导入，
+    # 这样形成的处理队列会造成资源浪费。因此，Lightning 稍微增大了前几个
+    # 区块的大小，从而合理分配资源。该参数也决定了向上扩展（scale up）因
+    # 数，代表在完全并发下“导入”和“写入”过程的持续时间比。这个值也可以通过
+    # 计算 1 GB 大小单张表的（导入时长/写入时长）得到。精确的时间可以在日志
+    # 里看到。如果 “导入”更快，区块大小差异就会更小；比值为 0 则说明区块大小
```
Comment (yikeke) [inline tools/lightning/deployment.md]:
```suggestion
    # 里看到。如果“导入”更快，区块大小差异就会更小；比值为 0 则说明区块大小
```

File: tools/lightning/deployment.md [hunk 12]
```diff
@@ -301,6 +320,27 @@ TiDB-Lightning 可随 TiDB 集群一起用 [Ansible 部署](../../op-guide/ansib
...
+    separator = ','
+    # 引用定界符，可为单个 ASCII 字符或空字符串。
+    delimiter = '"'
+    # CSV 文件是否包含表头。
+    # 如果为 true，第一行导入时会被跳过。
+    header = true
+    # CSV 是否包含 NULL。
+    # 如果 `not-null` 为 true，CSV 所有列都不能解析为 NULL。
+    not-null = false
+    # 如果 `not-null` 为 false（CSV 可以包含 NULL）。
```
Comment (yikeke) [inline tools/lightning/deployment.md]:
```suggestion
    # 如果 `not-null` 为 false（即 CSV 可以包含 NULL）。
```

File: tools/lightning/csv.md [hunk 13]
```diff
@@ -0,0 +1,209 @@
...
+# 引用定界符，可以为 ASCII 字符或空字符。
+delimiter = '"'
+# CSV 文件是否包含表头。
+# 如果为 true，首行将会被跳过。
+header = true
+# CSV 是否包含 NULL。
+# 如果为 true，CSV 文件的任何列都不能解析为 NULL。
+not-null = false
+# 如果 `not-null` 为 false（CSV 可以包含 NULL），
+# 为此值的字段即解析为空。
```
Comment (yikeke) [inline tools/lightning/csv.md]:
```suggestion
# 为以下值的字段将会被解析为 NULL。
```

File: tools/lightning/deployment.md [hunk 14]
```diff
@@ -301,6 +320,27 @@ TiDB-Lightning 可随 TiDB 集群一起用 [Ansible 部署](../../op-guide/ansib
...
+    # 为以下值的字段将会被解析为 NULL。
+    null = '\N'
+    # 是否解析字段内反斜线转义符。
+    backslash-escape = true
+    # 如果有行以分隔符结尾，删除尾部分隔符。 
+    trim-last-separator = false
+
+
```
Comment (yikeke) [inline tools/lightning/deployment.md]:
```suggestion
```
plz remove the extra line here

File: tools/lightning/errors.md [hunk 15]
```diff
@@ -44,6 +43,16 @@ Lightning 的正常速度为每条线程每 2 分钟导入一个 256 MB 的数
 1. 使用 `tidb-lightning-ctl --error-checkpoint-destroy=all` 把出错的表删除，然后重启 Lightning 重新导入那些表。
 2. 把断点存放在外部数据库（修改 `[checkpoint] dsn`），减轻目标集群压力。
 
+## Checkpoint for … has invalid status: 18
+
+**原因**: [断点续传](../../tools/lightning/checkpoints.md) 已启用。Lightning 或 Importer 之前发生了异常退出。为了防止意外数据损害，Lighting 在错误解决以前不会启动。
```
Comment (yikeke) [inline tools/lightning/errors.md]:
```suggestion
**原因**: [断点续传](../../tools/lightning/checkpoints.md)已启用。Lightning 或 Importer 之前发生了异常退出。为了防止数据意外损坏，Lighting 在错误解决以前不会启动。
```

File: tools/lightning/csv.md (hunk 7 above)
Comment (dcalvin) [reply tools/lightning/csv.md]: As confirmed with @kennytm, the previous translation is correct.

File: tools/lightning/csv.md [hunk 16]
```diff
@@ -0,0 +1,209 @@
...
+- 引用不影响字段是否为空。
+
+    例如有如下 CSV 文件：
+
+    ```csv
+    A,B,C
+    \N,"\N",
+    ```
+
+    在默认设置（`not-null = false; null = '\N'`）下，列 `A` and `B` 导入 TiDB 后都将会转换为 NULL。列 `C` 是空字符串，但并不解析为 NULL。
```
Comment (dcalvin) [inline tools/lightning/csv.md]:
```suggestion
    在默认设置（`not-null = false; null = '\N'`）下，列 `A` and `B` 导入 TiDB 后都将会转换为 NULL。列 `C` 是空字符串 `''`，但并不会解析为 NULL。
```

**Pull request merged**